libs/wingc/src/type_check.rs

```diff
@@ -1631,82 +1636,111 @@ impl<'a> TypeChecker<'a> {
 	/// # Returns
 	/// The hydrated type reference
 	///
-	fn hydrate_class_type_arguments(&mut self, env: &SymbolEnv, original_fqn: &str, type_param: TypeRef) -> TypeRef {
+	fn hydrate_class_type_arguments(
+		&mut self,
+		env: &SymbolEnv,
+		original_fqn: &str,
+		type_params: Vec<TypeRef>,
```

Comment:
Feels like this should be deduced from the @typeparam?

Reply:
The actual type of our collection types are the built-in `Types::Array` and such. This is where the type params are actually supplied.

```
let d: Array<num> = [1, 2, 3];
```

So when hydrating the JSII classes, we need to gather the args from that type info.